File: app/src/main/java/com/kunzisoft/keepass/database/cursor/EntryCursorV3.java
package com.kunzisoft.keepass.database.cursor;

import com.kunzisoft.keepass.database.element.PwDatabase;
import com.kunzisoft.keepass.database.element.PwEntryV3;

public class EntryCursorV3 extends EntryCursor<PwEntryV3> {

    public void addEntry(PwEntryV3 entry) {
        addRow(new Object[] {entryId,
                entry.getNodeId().getId().getMostSignificantBits(),
                entry.getNodeId().getId().getLeastSignificantBits(),
                entry.getTitle(),
                entry.getIcon().getIconId(),
                PwDatabase.UUID_ZERO.getMostSignificantBits(),
                PwDatabase.UUID_ZERO.getLeastSignificantBits(),
                entry.getUsername(),
                entry.getPassword(),
                entry.getUrl(),
                entry.getNotes()});
        entryId++;
    }

}

File: app/src/main/java/com/kunzisoft/keepass/database/cursor/EntryCursorV3.kt
package com.kunzisoft.keepass.database.cursor

import com.kunzisoft.keepass.database.element.PwDatabase
import com.kunzisoft.keepass.database.element.PwEntryV3

class EntryCursorV3 : EntryCursorUUID<PwEntryV3>() {

    override fun addEntry(entry: PwEntryV3) {
        addRow(arrayOf(
                entryId,
                entry.id.mostSignificantBits,
                entry.id.leastSignificantBits,
                entry.title,
                entry.icon.iconId,
                PwDatabase.UUID_ZERO.mostSignificantBits,
                PwDatabase.UUID_ZERO.leastSignificantBits,
                entry.username,
                entry.password,
                entry.url,
                entry.notes
        ))
        entryId++
    }

}
